tools/app_interface_reporter.py

```diff
@@ -495,33 +515,57 @@ def get_apps_data(date, month_delta=1):
+        logging.info(f"collecting dashdotdb information for {app_name}")
         app_namespaces = []
         for namespace in namespaces:
             if namespace['app']['name'] != app['name']:
                 continue
             app_namespaces.append(namespace)
-        app_metrics = {}
+        vuln_mx = {}
+        validt_mx = {}
         for family in text_string_to_metric_families(metrics):
             for sample in family.samples:
-                if sample.name != 'imagemanifestvuln_total':
-                    continue
-                for app_namespace in app_namespaces:
-                    cluster = sample.labels['cluster']
-                    if app_namespace['cluster']['name'] != cluster:
-                        continue
-                    namespace = sample.labels['namespace']
-                    if app_namespace['name'] != namespace:
-                        continue
-                    severity = sample.labels['severity']
-                    if cluster not in app_metrics:
-                        app_metrics[cluster] = {}
-                    if namespace not in app_metrics[cluster]:
-                        app_metrics[cluster][namespace] = {}
-                    if severity not in app_metrics[cluster][namespace]:
+                if sample.name == 'imagemanifestvuln_total':
+                    for app_namespace in app_namespaces:
+                        cluster = sample.labels['cluster']
+                        if app_namespace['cluster']['name'] != cluster:
+                            continue
+                        namespace = sample.labels['namespace']
+                        if app_namespace['name'] != namespace:
+                            continue
+                        severity = sample.labels['severity']
+                        if cluster not in vuln_mx:
+                            vuln_mx[cluster] = {}
+                        if namespace not in vuln_mx[cluster]:
+                            vuln_mx[cluster][namespace] = {}
+                        if severity not in vuln_mx[cluster][namespace]:
+                            value = int(sample.value)
+                            vuln_mx[cluster][namespace][severity] = value
+                if sample.name == 'deploymentvalidation_total':
+                    for app_namespace in app_namespaces:
+                        cluster = sample.labels['cluster']
+                        if app_namespace['cluster']['name'] != cluster:
+                            continue
+                        namespace = sample.labels['namespace']
+                        if app_namespace['name'] != namespace:
+                            continue
+                        validation = sample.labels['validation']
+                        # dvo: fail == 1, pass == 0, py: true == 1, false == 0
+                        # so: ({false|pass}, {true|fail})
+                        status = ('Passed', 'Failed')[int(sample.labels['status'])]
```

Comment:
@apahim  this was the hiccup causing the casting exception; added int(). ptal -- thanks!